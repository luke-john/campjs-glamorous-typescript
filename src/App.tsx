import * as React from 'react'

import {
    StyledApp,
    StyledHeader,
    StyledIntro,
    StyledLogo,
} from './App.glamorous'

// tslint:disable-next-line no-var-requires
const logo = require('./logo.svg')

class App extends React.Component<{}, {}> {
    render() {
        return (
            <StyledApp>
                <StyledHeader>
                    <StyledLogo
                        src={logo}
                        className="App-logo"
                        alt="logo"
                    />
                    <h2>Welcome to React</h2>
                </StyledHeader>
                <StyledIntro>
                    To get started, edit <code>src/App.js</code> and save to reload.
                </StyledIntro>
            </StyledApp>
        )
    }
}

export default App
