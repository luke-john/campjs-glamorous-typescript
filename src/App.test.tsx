import * as React from 'react'
import * as ReactDOM from 'react-dom'

import * as renderer from 'react-test-renderer'

import App from './App'

it('renders without crashing', () => {
    const div = document.createElement('div')

    ReactDOM.render(<App />, div)
})

it('renders expected things', () => {
    const tree = renderer.create(
       <App />,
    ).toJSON()

    expect(tree).toMatchSnapshot()
})
