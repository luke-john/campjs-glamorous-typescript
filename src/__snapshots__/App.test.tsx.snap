// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders expected things 1`] = `
.glamor-3,
[data-glamor-3] {
  text-align: center;
}

.glamor-1,
[data-glamor-1] {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.glamor-0,
[data-glamor-0] {
  animation: animation_oy05gt infinite 20s linear;
  height: 80px;
  -webkit-animation: animation_oy05gt infinite 20s linear;
}

.glamor-2,
[data-glamor-2] {
  font-size: large;
}

<div
  className="glamor-3"
>
  <div
    className="glamor-1"
  >
    <img
      alt="logo"
      className="glamor-0 App-logo"
      src="logo.svg"
    />
    <h2>
      Welcome to React
    </h2>
  </div>
  <p
    className="glamor-2"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;
